dnn/neural_network.py

```diff
@@ -20,6 +22,8 @@ def __init__(
         actfunc="sigmoid",
         costfunc="rss",
         testmode="classify",
+        optimize_method="adam",
+        adam_batch_size=2,
```

Comment:
```suggestion
        adam_batch_size=structure[-1],
```
としないとラベルの長さが２以外の時`adam_batch_size`を引数取らなくちゃいけないという困った仕様。
(structure is not defined)

ここ以外にも結構汚い